src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5260,6 +5260,28 @@ class Parser {
         ALLOC_TOKEN(token);
         while (true) {
             *token = this->lookahead;
+
+            if (UNLIKELY(token->type == Token::PunctuatorToken
```

Comment:
What is this parsing algorithm based on?

Reply:
I added comment

Reply:
Thanks :+1: